parser_library/src/processing/statement_fields_parser.cpp

```diff
@@ -59,16 +56,19 @@ std::pair<semantics::operands_si, semantics::remarks_si> statement_fields_parser
     bool after_substitution,
     semantics::range_provider field_range,
     processing::processing_status status,
-    const std::function<void(diagnostic_op)>& add_diag)
+    diagnostic_op_consumer& add_diag)
 {
     m_hlasm_ctx->metrics.reparsed_statements++;
 
-    parsing::parser_error_listener_ctx listener(
-        *m_hlasm_ctx, after_substitution ? &field : nullptr, add_diag, field_range);
-
     const auto original_range = field_range.original_range;
 
-    const auto& h = prepare_parser(field, after_substitution, std::move(field_range), status, listener);
+
+    diagnostic_op_consumer_transform add_diag_subst([&](diagnostic_op diag) {
+        if (after_substitution)
+            diag.message = "While evaluating the result of substitution '" + field + "' => " + diag.message;
+        add_diag.add_diagnostic(std::move(diag));
+    });
```

Comment:
```suggestion
    diagnostic_op_consumer_transform add_diag_subst([&field, after_substitution](diagnostic_op diag) {
        if (after_substitution)
            diag.message = "While evaluating the result of substitution '" + field + "' => " + std::move(diag.message);
        add_diag.add_diagnostic(std::move(diag));
    });
```
and re-run the benchmark?